cibuildwheel/logger.py

```diff
@@ -222,6 +275,68 @@ def _fold_group_identifier(name: str) -> str:
         # lowercase, shorten
         return identifier.lower()[:20]
 
+    def _github_step_summary(self, duration: float, options: "Options") -> str:
+        """
+        Returns the GitHub step summary, in markdown format.
+        """
+        out = io.StringIO()
+        options_summary = options.summary(
+            identifiers=[bi.identifier for bi in self.summary], skip_unset=True
+        )
+        out.write(
+            textwrap.dedent("""\
+                ### 🎡 cibuildwheel
+
+                <details>
+                <summary>
+                Build options
+                </summary>
+
+                ```yaml
+                {options_summary}
+                ```
+
+                </details>
+
+            """).format(options_summary=options_summary)
+        )
+
+        out.write(
+            textwrap.dedent("""\
```

Comment:
You can use `inspect.cleandoc` instead of `textwrap.dedent` then you don't have to worry about leading/trailing whitespace. Up to you which you prefer.
```suggestion
            inspect.cleandoc("""
```